fastcrawler/core/spider.py

```diff
@@ -17,39 +24,62 @@ class Spider:
     that is connected to current spider class
     """
 
-    instances: list["Spider"]
     engine_request_limit: int | None = None
-    parser: ParserProtocol
+    parser: Type[ParserProtocol] | None
     start_url: set[str] | _Depends
-    data_model: BaseModel | None = None
-    depth: int = 1_000
+    data_model: Type[BaseModel] | None = None
+    max_depth: int | None = None
     _is_stopped = False
+    batch_size: int | None = None
+    request_sleep_interval: float | None = None
+    cycle_sleep_interval: float | None = None
+    max_request_count: int | None = None
 
     def __init__(
         self,
-        engine: None | EngineProto = None,
-        parser: ParserProtocol | None = None,
-        model: BaseModel | None = None,
+        engine: None | Type[EngineProto] = None,
+        parser: Type[ParserProtocol] | None = None,
+        model: Type[BaseModel] | None = None,
     ):
-        self._engine: EngineProto = engine or AioHttpEngine
+        self._engine: Type[EngineProto] = engine or AioHttpEngine  # type: ignore
+
+        if model is None and self.data_model is None:
+            raise SpiderBadConfiguraton("A data model must be specified")
+
+        if self.engine_request_limit is None and self._engine.default_request_limit is None:
+            raise SpiderBadConfiguraton("Request limit must be specified")
+
         self.parser = parser or HTMLParser
-        self.data_model = model or self.data_model
+        self._data_model: Type[BaseModel] = model or self.data_model  # type: ignore
         self._crawled_urls = set()
         self._pending_urls = set()
-        self.engine_request_limit = self.engine_request_limit or self._engine.default_request_limit
+        self._engine_request_limit: int = (
+            self.engine_request_limit or self._engine.default_request_limit
+        )
+        self.request_sent = 0
 
     @property
-    def engine(self) -> EngineProto:
+    def engine(self) -> Type[EngineProto]:
         """Method to access engine"""
         return self._engine
 
     @property
-    def is_stopped(self) -> set:
+    def instances(self) -> list[Self | "Spider"]:
+        if not hasattr(self, "_instances"):
+            self._instances = [self]
+        return self._instances
+
+    @instances.setter
```

Comment:
bad naming

Reply:
what do you mean? the instances?
any suggestion?
Like linked_spiders?

Reply:
spiders maybe? i don't know you decide